pageserver/src/tenant/timeline.rs

```diff
@@ -535,6 +550,21 @@ impl From<GetVectoredError> for CreateImageLayersError {
     }
 }
 
+impl From<GetReadyAncestorError> for PageReconstructError {
+    fn from(e: GetReadyAncestorError) -> Self {
+        match e {
+            GetReadyAncestorError::AncestorStopping(tid) => {
+                PageReconstructError::AncestorStopping(tid)
+            }
+            GetReadyAncestorError::AncestorLsnTimeout(wait_err) => {
+                PageReconstructError::AncestorLsnTimeout(wait_err)
+            }
+            GetReadyAncestorError::Cancelled => PageReconstructError::Cancelled,
+            GetReadyAncestorError::Other(other) => PageReconstructError::Other(other),
+        }
```

Comment:
Helps with typing if one `use EnumType::*;`:

```
        use GetReadyAncestorError::*;

        match e {
            AncestorStopping(tid) => {
                PageReconstructError::AncestorStopping(tid)
            }
            AncestorLsnTimeout(wait_err) => {
                PageReconstructError::AncestorLsnTimeout(wait_err)
            }
            Cancelled => PageReconstructError::Cancelled,
            Other(other) => PageReconstructError::Other(other),
        }
```

Lines are probably still too long though.

Reply:
You're right. It formats nicer. Thanks: https://github.com/neondatabase/neon/pull/6543/commits/63209c682065936e789d7f05851fbe53c8898d42